### ISSUE TITLE
Race condition with capitalization

### ISSUE THREAD

**Issue by user202729** [openstenoproject/plover#1051]
## Summary

Sometimes a noncapitalized character is wrongly capitalized when it's before a upper case character.

## Reproducing

Pressing "P-BG" a few times gives:

    pK pK pK PK pK pK pK PK pK PK PK pK pK pK PK PK pK pK pK pK pK pK pK pK PK PK

## Plover Version

1ff8bb95d834cffc8edb1ecde3cd70857e0f236a

## System

Arch Linux.

**Comment by SeaLiteral** [comment]
Which window manager are you using? I don't get this issue on Unity on Ubuntu when writing pH, {Mc^}{-|} and pK, two lines of each.

I remember there being an issue with ibus. Back then I could steno into gedit and it would work fine, but if I stenoed into Firefox it would miss some characters in the output. Plover was outputting them, but Firefox wasn't receiving them. Maybe you could try writing strings of pK into different applications and see if this happens in all of them or just some of them.

**Comment by user202729** [comment]
I'm using cinnamon.
I can't reproduce it in xterm, only in my browser and Plover lookup window.

**Comment by SeaLiteral** [comment]
Does it also happen in other Plover windows, such as the add translation dialog? You should be able to open that from the main window.

And which browser is it? Firefox? Chromium? Konqeror? Have you tried stenoing into a text editor or word processor?

**Comment by user202729** [comment]
1. It does happen in the add translation dialog.
2. qutebrowser.
3. Can't reproduce in vim or gvim. Probably an issue with Qt or PyQt.

**Comment by benoit-pierre** [comment]
AFAIK, neither gvim nor xterm use ibus, so do you have ibus running?

**Comment by user202729** [comment]
```
$ ibus version
bash: ibus: command not found
```

I think I don't install it.

**Comment by benoit-pierre** [comment]
Can you reproduce the issue with xdotool? Using for example: `xdotool type --delay 0 'pK pK pK pK'`.

**Comment by user202729** [comment]
No, regardless of whether plover is enabled or disabled.

**Comment by benoit-pierre** [comment]
That was without Plover running. Anyway, you could check if the sequence of events send by Plover is correct, although it it works in other applications, it probably is. Use `xinput test-xi2 --root DEVID` (replace `DEVID` with the device number for `Virtual core XTEST keyboard` (`xinput list` to get it)). How did you install Plover? What about PyQt5? Your using a custom version, can you confirm this happen with master too?

**Comment by SeaLiteral** [comment]
Wouldn't xdotool type pK four times in xterm? If the issue doesn't happen when using Plover in xterm, it probably doesn't happen when using xdotool in xterm.

I don't have this problem myself, but I just got Python to use xdotool to write a string of pKs in Firefox:

```import os
import time

def test_writing(x):
    time.sleep(2)
    os.system("xdotool type --delay 0 '"+x+"'")

test_writing('pK '*6)```

The output I get is `pK pK pK pK pK pK `.

**Comment by benoit-pierre** [comment]
xdootool will send events to the focused window, so just use sleep before the command and quickly select the problematic application: `sleep 3; xdotool ..`

**Comment by user202729** [comment]
It does happen to master.

**Comment by user202729** [comment]
Sequence of event looks correct.

However if  that's the case, the bug is probably not with Plover, but PyQt/Qt.

**Comment by benoit-pierre** [comment]
Yes, have you tried again with xdotool, but focusing the correct application? So you still have not mentioned how Plover/PyQt5 were installed.

**Comment by user202729** [comment]
xdotool was tried on Plover lookup window.
Plover is from git. PyQt5 is from arch linux package.

**Comment by SeaLiteral** [comment]
I don't get this issue on Cinnamon on Ubuntu. So this could be specific to Arch Linux.

**Comment by mkrnr** [comment]
Closing this ticket due to its age. Please comment if this is still an issue.
